config/fetcher_test.go

```diff
@@ -0,0 +1,11 @@
+package config_test
+
+import "github.com/stretchr/testify/mock"
+
+type MockFetcher struct {
+	mock.Mock
+}
+
+func (m *MockFetcher) Get(key string) (val string) {
+	return m.Called(key).String(0)
```

Comment:
This is a case where a fetcher that reads from a `map[string]string` would make more sense IMO.

```golang
env := EnvironmentOf(NewFetcher(map[string]string{"FOO": "FOO"}))
got := c.GotFn(env, c.Val)

if c.Expected != got {
  t.Errorf("....")
}
```

I just don't see what mocking gets us here.